libopae/src/enum.c

```diff
@@ -74,9 +74,13 @@ struct dev_list {
 	struct dev_list *fme;
 };
 
-static bool matches_filter(const struct dev_list *attr,
-			   const fpga_properties filter)
+bool matches_filter(const struct dev_list *attr, const fpga_properties filter)
 {
+	if (!filter) {
```

Comment:
This is caught by matches_filters() in the case that num_filters is 0.  (there is logic that makes sure that filters is NULL if num_filters is 0 just inside fpgaEnumerate.  It is not technically an error for filter to be NULL.  That is the "match all" case.  You're right that filter mustn't be NULL here, but the code is structured so that we don't need to check.